add_normweights.py

```diff
@@ -105,40 +123,77 @@ def expand_ttrees(args):
   debug = args.debug
   ttree_name = args.ttree_name
   dry_run = args.dry_run
+  ncpu = args.ncpu
 
-  # Create logger
+  # get list of files
+  file_list = handleInput(input_dir)
+  # remove already expanded files
+  file_list = [i for i in file_list if '_expanded.root' not in i]
+
+  # update global logger to new debugging
   logging.basicConfig(level = 'INFO' if not debug else 'DEBUG', format = '%(levelname)s: %(message)s')
   log = logging.getLogger()
   
   log.info('Files in the following directory will be merged: {}'.format(input_dir))
 
   # Get sum of weights from all input files (by DSID)
   sum_of_weights = {} # sum of weights for each dsid
-  for file_name in os.listdir(input_dir):
+  for file_name in file_list:
     # Make sure it's a ROOT file
     if not file_name.endswith('.root'): continue
     # Get metadata from all files, even those w/ empty TTrees
-    full_file_name = '{}/{}'.format(input_dir, file_name)
     try:
-      tfile = TFile.Open(full_file_name)
+      tfile = TFile.Open(file_name)
     except OSError:
-      raise OSError('{} can not be opened'.format(full_file_name))
+      raise OSError('{} can not be opened'.format(file_name))
     # Identify DSID for this file
-    dsid = int(input_dir.split('user.')[1].split('.')[2])
-    log.debug('dsid: {} identified from input_dir: {}'.format(dsid, input_dir))
+    dsid = int(file_name.split('user.')[1].split('.')[2])
+    log.debug('dsid: {} identified from input_dir: {}'.format(dsid, file_name))
     # Get sum of weights from metadata
     metadata_hist = tfile.Get('MetaData_EventCount')
     if dsid not in sum_of_weights:
       sum_of_weights[dsid] = metadata_hist.GetBinContent(3) # initial sum of weights
     else:
       sum_of_weights[dsid] += metadata_hist.GetBinContent(3) # initial sum of weights
+  
+  log.info('Sum of weights: {}'.format(sum_of_weights))
+
+  # create job configurations
+  config = []
+  for file_name in file_list:
+    config.append({
+      "file_name" : file_name, 
+      "input_dir" : input_dir,
+      "ttree_name" : ttree_name,
+      "output_dir" : output_dir,
+      "dry_run" : dry_run,
+      "pmg_file_name" : pmg_file_name,
+      "sum_of_weights" : sum_of_weights,
+      "debug" : debug
+    })
 
   # Add normweight and write a new ROOT file
-  for file_name in os.listdir(input_dir):
-    add_normweight(file_name, input_dir, ttree_name, output_dir, dry_run, pmg_file_name, sum_of_weights, log, debug)
-
+  if not dry_run:
+    if ncpu > 1:
+      results = Pool(ncpu).map(add_normweight, config)
+    else:
+      for conf in config:
+        add_normweight(conf)
+  
   log.info('>>> All done <<<')
 
+def handleInput(data):
```

Comment:
Everything looks good, I would just harmonize the spacing and use 4 spaces everywhere. But that could be done in a separated PR.

Reply:
done I think